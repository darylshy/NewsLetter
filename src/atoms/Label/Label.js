import React from 'react';
import classes from './Label.module.scss';
import PropTypes from 'prop-types';

const Label = props => {
  return <div className={classes.Label}>{props.label}</div>;
};

Label.defaultProps = {
  label: 'Label'
};

Label.propTypes = {
  label: PropTypes.string.isRequired
};

export default Label;
