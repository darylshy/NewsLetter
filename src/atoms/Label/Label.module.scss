@import "../../styles/global/globals.scss";
.Label {
  font-family: "Elaine_D02";
  color: $font-green;
  font-size: $font-size-default;
  @include desktop {
    font-size: 6rem;
  }
  @include tabletL {
    font-size: 6rem;
  }
  @include tabletP {
    font-size: 5rem;
  }
  @include mobile {
    font-size: 3.6rem;
  }
}
