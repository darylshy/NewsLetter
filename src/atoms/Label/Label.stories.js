import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, text } from '@storybook/addon-knobs/react';
import React from 'react';

import Label from './Label';

storiesOf('Atoms/Label', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add('Standard Label Element', () => <Label label={text('label')} />, {
    info: {
      text: 'Label Element'
    }
  });
