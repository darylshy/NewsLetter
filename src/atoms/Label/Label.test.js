import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Label from './Label';
import React from 'react';

configure({ adapter: new Adapter() });

describe('<Label />', () => {
  let component;
  beforeEach(() => {
    component = shallow(<Label label="label" />);
  });
  it('should render an Label component', () => {
    expect(component).toMatchSnapshot();
  });
  it('should find a div with className "Label"', () => {
    expect(component.find('div').hasClass('Label')).toEqual(true);
  });
});
