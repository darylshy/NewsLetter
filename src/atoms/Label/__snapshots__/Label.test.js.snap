// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render an Label component 1`] = `ShallowWrapper {}`;
